### PR TITLE
Add FCM notifications

### PR DESCRIPTION
**Pull request by khbnr**
Notifications will be sent either to an HTTP endpoint or as Google FCM, depending on the format of  the registered notificationUrl

### CODE REVIEW

File: src/main/kotlin/com/healthmetrix/labres/notifications/Notification.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        fun from(notificationUrl: String) = when {
+            notificationUrl.isHttp() -> HttpNotification(notificationUrl)
+            notificationUrl.isLabresFcmUrl() -> FcmNotification(notificationUrl.asFcmToken())
+            else -> null
+        }
+
+        private fun String.isHttp() =
+            this.startsWith(HTTP_SCHEME, true) || this.startsWith(HTTPS_SCHEME, true)
+
+        private fun String.isLabresFcmUrl() = this.startsWith(LAB_RES_FCM_SCHEME)
```
Comment (varunjayaraman) [inline src/main/kotlin/com/healthmetrix/labres/notifications/Notification.kt:21]: nit: `isLabResFcmUrl()`.

If the constant val is `LAB_RES_FCM_SCHEME` then i think the `camelCasing` should reflect that they are separate words too

File: src/main/kotlin/com/healthmetrix/labres/notifications/NotificationsConfig.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        val options = FirebaseOptions.builder()
+            .setCredentials(credentials)
+            .build()
+
+        val app = FirebaseApp.initializeApp(options)
+
+        val messaging = FirebaseMessaging.getInstance(app)
+
+        val dryRun = stage == "local"
```
Comment (varunjayaraman) [inline src/main/kotlin/com/healthmetrix/labres/notifications/NotificationsConfig.kt:49]: does it make sense to make the `dryRun` configurable from `application.yaml`?

File: src/test/kotlin/com/healthmetrix/labres/notifications/ManualNotifierTest.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    /*
+        To test sending notifications manually, you need the following prerequisites:
+        - AWS credentials having the permission to check out the necessary secrets for FCM and basic auth locally set up
+        - env NOTIFICATION_TARGET set to where ever you want to send a notification to
+        - Comment out the @Disabled annotation on the test
+
+        Then you can run this test and invoke actual notifications being sent
+     */
+    @Test
+    fun `send test notification`() {
```
Comment (varunjayaraman) [inline src/test/kotlin/com/healthmetrix/labres/notifications/ManualNotifierTest.kt:44]: does it make sense to implement this as a gradle task instead?

File: src/main/kotlin/com/healthmetrix/labres/notifications/NotificationsConfig.kt (hunk 2 above)
Comment (khbnr) [reply src/main/kotlin/com/healthmetrix/labres/notifications/NotificationsConfig.kt:49]: I'm trying to think of situations where we actually need this to be false besides the initial testing. We already have the `notify` profile in place to toggle the feature of sending notifications or just log them. If I'd add the dryRun flag to the `application.yaml` I would probably set it to true by default and only to false on the `notify` profile. But if this one isn't enabled, the code won't be added to the spring context anyways.
I also can't use the `FcmNotifier` always and toggle only the `dryRun` flag based on the `notify` profile, cause then the instantiation of the FcmNotifier would probably fail for not having a valid Google Cloud service account credentials JSON as secret. 
So long story short, I don't see too much value having the flag at all and not just hard coding it to always false. Would leave it like it is tho in case someone needs to ever change the notification stuff and wants a quite easy way for testing it first locally.

File: src/test/kotlin/com/healthmetrix/labres/notifications/ManualNotifierTest.kt (hunk 3 above)
Comment (khbnr) [reply src/test/kotlin/com/healthmetrix/labres/notifications/ManualNotifierTest.kt:44]: I wouldn't know a proper way to get a partial spring context up and invoke a method on a bean there via gradle without having to add more stuff than just this test. 
I know it would be cleaner, but on the other hand I don't see us using this test a lot of times except some testing with the Android app once or twice. I would rather therefore remove this file instead of putting more effort into it.

File: src/main/kotlin/com/healthmetrix/labres/notifications/NotificationsConfig.kt (hunk 2 above)
Comment (varunjayaraman) [reply src/main/kotlin/com/healthmetrix/labres/notifications/NotificationsConfig.kt:49]: alright

File: src/test/kotlin/com/healthmetrix/labres/notifications/ManualNotifierTest.kt (hunk 3 above)
Comment (varunjayaraman) [reply src/test/kotlin/com/healthmetrix/labres/notifications/ManualNotifierTest.kt:44]: kk
